infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/fsm/InSnapshotSyncState.java

```diff
@@ -123,6 +125,7 @@ public LogReplicationState processEvent(LogReplicationEvent event) throws Illega
                     // Re-trigger SnapshotSync due to error, generate a new event Id for the new snapshot sync
                     LogReplicationState inSnapshotSyncState = fsm.getStates().get(LogReplicationStateType.IN_SNAPSHOT_SYNC);
                     UUID newSnapshotSyncId = UUID.randomUUID();
+                    SnapshotSyncUtils.modifyUpgradeSnapshotSyncEvent(transitionEventId, newSnapshotSyncId, new CorfuStore(fsm.getAckReader().getRuntime()));
```

Comment:
what is the purpose of this change?

Reply:
After the rolling upgrade force sync event has been process by the event listener the following ENFORCE_SNAPSHOT_SYNC is the same as any other, so if it happens to be cancelled and re-triggered we would have no way to know that the uuid of the new snapshot sync was for upgrade when we attempt to remove the event from the event table.